main.go

```diff
@@ -75,61 +75,81 @@ func main() {
 func startManager() error {
 	logger.Info("Starting k8s-service-discovery...")
 
-	watchNamespace, err := readWatchNamespace()
+	watchNamespace, err := config.ReadWatchNamespace()
 
 	options := getK8sManagerOptions(watchNamespace)
```

Comment:
Shouldn't the error be catched before we get the K8sManagerOptions? The error is thrown beforehand by reading the environment variable